### DISCUSSION TITLE
Cross OS Compat

### DISCUSSION THREAD

**Discussion by Lewiscowles1986** [jezek/xgb#15] [Q&A] [answered]
Is it known if there are any libraries which use this, which can be used as a wrapper, to avoid having a linux with x-windows only build of software?

**Comment by jezek** [comment]
I don't fully understand, what you're asking. You're looking for some library, that can be used with wayland?

Maybe this is a page for you. https://github.com/go-graphics/go-gui-projects

**Comment by Lewiscowles1986** [comment]
I'm not asking for wayland. I was wondering if anyone uses `xgb` with any other libs that help to deliver non x-windows only support for a similar API. It's okay for no to be the answer. I Really included it as a community comment; but you don't have a community tab.

**Reply by jezek** [reply]
>  I Really included it as a community comment; but you don't have a community tab.

Comment tab created, question trasferred.

> I was wondering if anyone uses xgb with any other libs that help to deliver non x-windows only support for a similar API.

I still don't understand what do you mean by this. Especially the `deliver non x-windows only support for a similar API` part. Please explain, if possible with an example.

**Reply by Lewiscowles1986** [reply]
In order to make progress here, it might be easier if I ask you some questions, so that I can ensure we have a stable base.

Is XGB only for X-Windows (X11) based systems? (x-windows)
Does XGB have a stable set of functions exported for those using? (API)

**Reply by jezek** [reply]
> Is XGB only for X-Windows (X11) based systems? (x-windows)

As far I know, yes. XGB stands for X protocol Go-language Bindings, which is something like [XCB](https://xcb.freedesktop.org/) and most of the functions in this package are generated using [xproto](https://x.org/releases/X11R7.7/doc/xproto/x11protocol.html). So it is designed especialy for X11. You can use it on wayland when you have [Xwayland](https://wayland.freedesktop.org/xserver.html).

>  Does XGB have a stable set of functions exported for those using? (API)

I think you mean the xproto thing, right? Or anything else?

**Reply by Lewiscowles1986** [reply]
So what I'm really looking for is something to map the intentions, to capabilities. It's cool that xgb only handles X protocol; but I'm looking for a bridging / facade library which will know the go platform, and be able to switch between X, and OSx and Windows. It might not exist (although plenty of cross-platform Go GUI apps exist)

**Reply by jezek** [reply]
I would use [SDL2](https://www.libsdl.org/) and [go-sdl2](https://github.com/veandco/go-sdl2). It is multiplatform and you can handle graphics, inputs, audio, video, etc.

**Comment by Lewiscowles1986** [comment]
Thanks for creating this space. It is better than issues when it exists. I did not notice or see it before.

**Reply by jezek** [reply]
> ...I did not notice or see it before.

You couldn't even if you wanted to, It wasn't created that time. I created it on behalf of your suggestion. ;)

**Comment by ghost** [comment, marked as answer]
https://github.com/go-vgo/robotgo can be used to control keyboard, mouse and capture the state of the screen. its cross platform.
